### PR TITLE
no_tracing decorator

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it adds `no_tracing` decorator which allows to skip tracing of simple resolvers we don't want.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/core/tracing.py [hunk 1]
```diff
@@ -5,10 +5,22 @@
 from graphql import ResolveInfo
 
 DEFAULT_RESOLVERS = {default_resolver, GlobalID.id_resolver}
+NO_TRACE_ATTR = "__no_trace"
+
+
+def no_trace(func):
+    def wrapper(*args, **kwargs):
+        info = next(arg for arg in args if isinstance(arg, ResolveInfo))
+        setattr(info.context, NO_TRACE_ATTR, True)
```
Comment (tomaszszymanski129) [inline saleor/core/tracing.py:14]: investigated and this propagates to other fields after first decorated field is resolved, will have to be changed.

**Comment by patrys** [comment]
Task linked: [SALEOR-2157 Add no_tracing decorator to resolvers](https://app.clickup.com/t/2549495/SALEOR-2157)

**Review by fowczarek — CHANGES_REQUESTED** [review]
I found a bug. 
My query:
```
{
  user(id: "VXNlcjo5NDI=") {
    id
    events {
      id
      message
      count
      user {
        email
      }
      orderLine{
        id
      }
    }
  }
}
```
User have 2 events assigned with `parameters`:
```
{
    "message": "aaa",
    "count": 5,
    "order_line_pk": 1
}
```

For your branch, I got 1 span for `CustomerEvent.message` but 0 spans for `CustomerEvent.orderLine` and `CustomerEvent.user` 
image

For master, I got 2 spans for `CustomerEvent.message`, `CustomerEvent.count`,  `CustomerEvent.orderLine` and `CustomerEvent.user` 
image


After your changes, we should receive 2 spans for `CustomerEvent.orderLine`, `CustomerEvent.user` 

To reproduce this case you should add

**Comment by tomaszszymanski129** [comment]
As discussed days ago, this will have to be handled different way, closing this one for now.

**Pull request closed without merging**